### PR TITLE
Fixed indexer TorrentSyndikat.cs - Adding wildcard to season search string

### PR DESCRIPTION
**Pull request by Seriox**
When searching for a season (manual search --> i.e. "**West World S03**") the indexer for TorrentSyndikat adds a "+" to every word. (AND Operator).
The search string then looks like this --> "**+West +World +S03**"
Unfortunately searching for  "+West +World +S03" doesn't yield any results on TS altough West World S03**E01** is available. 
Thats beacause a wildcard operator needs to be added to "S03" so it looks like this "**+West +World +S03***"

The code was revised to add a wildcard operator IF the search string ends with S01 / S02 / S03 and so on. Matching with RegEx "**s[0-9]{2}$**"

Is that an expected behaviour? Sonarr for example only has an automated search for "West World S03". (Without episodes) 

I'm new to to Github (not to coding though), so if something's missing please let me know.

### CODE REVIEW

**Comment by cadatoiva** [comment]
When Sonarr only searches for "S03" it's trying to find a season pack with all episodes in it, not individual episodes as far as I know. So the behavior of ignoring individual episodes within the search is expected, if I'm not mistaken.

**Comment by Seriox** [comment]
According to 

> When Sonarr only searches for "S03" it's trying to find a season pack with all episodes in it, not individual episodes as far as I know. So the behavior of ignoring individual episodes within the search is expected, if I'm not mistaken.

Yeah I'm still not sure if thats an expected behaviour or not. So let's say I add a series to Sonarr. Sonarr starts searching for S03. Currently there's only S03E01 availabe. So that means I get 0 results back from Jackett. 
According to [this reddit post](https://old.reddit.com/r/sonarr/comments/fkq1zb/help_downloading_single_episodes_instead_of/) "it’s expected that the indexer will return those results already."

**Comment by cadatoiva** [comment]
Well, Markus-101 is basically the lead developer over there, so I'll take his input as authoritative. However, there are API's in Jackett to make what you're doing much easier than how you've done it. If you checked "allow members write access" or however it's named, I'll adjust your code and push accordingly.

**Comment by Seriox** [comment]
Sweet, thanks a lot! I've just connected my Sonarr to the my local Jackett fork and it seems to be working. It's a bit of hassle to test this though. (Revers proxy with Fiddler...) 
I've just noticed that I missed setting RegEx to case insensitve.  I'm updating this pull request in a second.

**Comment by cadatoiva** [comment]
No need for Regex at all with the change. The query object's function `GetQueryString()` returns `SanitizedSearchString + " " + GetEpisodeSearchString()`

Jackett already parses out season and episode numbers found within a search string and saves them as numbers, parsing them back out into a standard S00E00 format when passing in the query string. Always capital, always at the end. We can use this to our advantage by splitting the Sanitized string once, and checking the Episode string for S without E and adding * to the end. (some shows are dailies, and the episode string also parses that as yyyy.MM.dd in the same place, so we have to make sure it's a true S00E00 string first)

**Comment by cadatoiva** [comment]
Please test my changes and make sure they work as intended.

**Comment by Seriox** [comment]
Unfortunately it's not working as intended. 
The Season information gets lost when creating the search string. For example "SeriesTitle S03" gets shrinked to just "SeriesTitle".

Search string entered in manual search --> "Westworld S03"
Search string from Jackett log file --> browse.php?incldead=1&rel_type=0&searchin=title&search=%2BWestworld

search_string2


search_string3

File: src/Jackett.Common/Indexers/TorrentSyndikat.cs [hunk 1]
```diff
@@ -146,20 +144,30 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+                        RegexOptions.Compiled).Select(term => $"+{term}");
+
+                    // If only season search add * wildcard to get all episodes
+                    if (query.IsTVSearch)
+                    {
+                        var tvEpisode = query.GetEpisodeSearchString();
+                        if (!string.IsNullOrWhiteSpace(tvEpisode) && tvEpisode.StartsWith("S") && !tvEpisode.Contains("E"))
+                            tvEpisode += "*";
+                        searchStringArray = searchStringArray.Append($"+{tvEpisode}");
+                    }
```
Comment (cadatoiva) [inline src/Jackett.Common/Indexers/TorrentSyndikat.cs]:
```suggestion
                    var tvEpisode = query.GetEpisodeSearchString();
                    if (!string.IsNullOrWhiteSpace(tvEpisode) && tvEpisode.StartsWith("S") && !tvEpisode.Contains("E"))
                        tvEpisode += "*";
                    searchStringArray = searchStringArray.Append($"+{tvEpisode}");
```

**Comment by cadatoiva** [comment]
That should fix it. Instead of only applying the change on a TV search specifically, it will check and apply changes any time the season section isn't blank. Test again please.

**Comment by Seriox** [comment]
That looks good! I've successfully tested a few cases. No issues so far. Connection with Sonarr seems fine as well. 
Just a quick thought: What would happen if Jackett is searching for a series called "SeriesS01Title S01"? (The string "S01" is part of the title?)
Thanks a lot! :)

**Comment by cadatoiva** [comment]
This is the code that parses the search terms, so you can follow that, looks like it will strip out the first instance of case sensitive S00 or S0000. https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Models/DTO/ApiSearch.cs#L12-L67

**Pull request merged**